lib/tasks/heroku.rake

```diff
@@ -70,6 +70,45 @@ namespace :heroku do
     Rails.logger.info("Done setting up Heroku review app DNS")
   end
 
+  task review_app_predestroy: :environment do
+    # Delete this app's hostnames from Route 53
+    gyr_hostname = MultiTenantService.new(:gyr).host
+    ctc_hostname = MultiTenantService.new(:ctc).host
+    Rails.logger.info("Deleting Route 53 DNS: gyr_hostname=#{gyr_hostname} ctc_hostname=#{ctc_hostname}")
+
+    heroku_app_name = ENV["HEROKU_APP_NAME"]
+    heroku_client = PlatformAPI.connect_oauth(ENV["HEROKU_PLATFORM_KEY"])
+    route53_client = Aws::Route53::Client.new(
+      access_key_id: ENV["HEROKU_DNS_AWS_ACCESS_KEY_ID"],
+      secret_access_key: ENV["HEROKU_DNS_SECRET_ACCESS_KEY"],
+      region: 'us-east-1',
+    )
+    [gyr_hostname, ctc_hostname].each do |hostname|
+      cname_target = heroku_client.domain.info(heroku_app_name, hostname)["cname"]
+      Rails.logger.info("Deleting AWS CNAME with cname_target=#{cname_target} fully_qualified_domain=#{hostname}")
+
+      route53_client.change_resource_record_sets(
```

Comment:
Per https://docs.aws.amazon.com/cli/latest/reference/route53/change-resource-record-sets.html :

> To delete a resource record set, you must specify all the same values that you specified when you created it.

Seems like a lot of work to delete something, but OK.